source/common/quic/envoy_quic_client_session.h

```diff
@@ -130,7 +130,7 @@ class EnvoyQuicClientSession : public QuicFilterManagerConnectionImpl,
   Stats::Scope& scope_;
   bool disable_keepalive_{false};
   Network::TransportSocketOptionsConstSharedPtr transport_socket_options_;
-  OptRef<QuicClientTransportSocketFactory> transport_socket_factory_;
+  OptRef<QuicTransportSocketFactoryBase> transport_socket_factory_;
```

Comment:
I'm mildly surprised that there are no APIs in `QuicClientTransportSocketFactory` that are required here. But I guess there aren't! LGTM.